include/dt-bindings/interrupt-controller/arm-gic.h

```diff
@@ -18,6 +18,12 @@
 #define	IRQ_TYPE_LEVEL		0x0
 #define	IRQ_TYPE_EDGE		0x1
 
+#define	GIC_SPI			0x0
+#define	GIC_PPI			0x1
+
 #define IRQ_DEFAULT_PRIORITY	0xa
 
+#define GIC_FIX_INTID(id)	((id##_TYPE == GIC_SPI) ? \
+					((id) + 32) : ((id) + 16))
```

Comment:
I think this should provide a cleaner and much easier to use interface:
```
#define GIC_PPI_BASE 16
#define GIC_SPI_BASE 32

#define GIC_PPI(id) (GIC_PPI_BASE + id)
#define GIC_SPI(id) (GIC_SPI_BASE + id)
```

With this, in dts, we can just do:
```
interrupts = <GIC_SPI(36) IRQ_DEFAULT_PRIORITY
		IRQ_TYPE_LEVEL>, ...
```

Also we don't need dts_fixup for every GIC mapped interrupt this way.

Reply:
Yes. I chose that way to be closer to how that is done in the Linux kernel but I like the fact that we can avoid the dts_fixup.